prisma/migrations/20210428060853_user_username_email_required_unique/migration.sql

```diff
@@ -0,0 +1,22 @@
+/*
+  Warnings:
+
+  - A unique constraint covering the columns `[username]` on the table `users` will be added. If there are existing duplicate values, this will fail.
+  - A unique constraint covering the columns `[email]` on the table `users` will be added. If there are existing duplicate values, this will fail.
+  - Made the column `name` on table `users` required. This step will fail if there are existing NULL values in that column.
+  - Made the column `username` on table `users` required. This step will fail if there are existing NULL values in that column.
+  - Made the column `password` on table `users` required. This step will fail if there are existing NULL values in that column.
+  - Made the column `email` on table `users` required. This step will fail if there are existing NULL values in that column.
+
+*/
+-- AlterTable
+ALTER TABLE "users" ALTER COLUMN "name" SET NOT NULL,
+ALTER COLUMN "username" SET NOT NULL,
+ALTER COLUMN "password" SET NOT NULL,
```

Comment:
This migration will probably fail. Currently password is set only after the user has signed up, through the email you receive. Unless the sign up process is changed to include the password, this column should be kept optional.

Reply:
Oh another thing, there's a bunch of users with duplicated emails and usernames, so the `unique` constraint will fail if we try to apply it. See the links I will send on discord for the sql queries I used to check.